### PR TITLE
Upgrade for Fly@2

### PR DESCRIPTION
**Pull request by lukeed**
Even though `Fly@2` is still in beta, I wanted to get this plugin ready for official release.

Changes:

* You may now pass an `entry` value to your options. Doing so will **override** / take precedence over all `files` matched via `source()`.

  This is useful for long task chains; 

  ```js
  yield this.source('src/**/*.js').xo().rollup({
    rollup: {entry: 'src/app.js'}
  }).uglify().target('dist/js');
  ```

* Replaced ESLint with XO
* Replaced AVA with Tape
* Reworked tests, within a true Fly environment

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,21 +31,20 @@
...
-    "eslint-config-standard": "^5.3.1",
-    "eslint-plugin-promise": "^1.3.2",
-    "eslint-plugin-standard": "^2.0.0",
-    "object-assign": "^4.1.0",
+    "fly": "beta",
     "rollup": "^0.34.1",
-    "rollup-plugin-babel": "^2.6.1"
-  },
-  "peerDependencies": {
-    "fly": "^1.4.1"
```
Comment (frantic1048) [inline package.json:45]: Why not keep a peerDep for fly ?

**Comment by lukeed** [comment]
When it comes time to publish, you can reestablish Fly as a peer dependency. For now, since `beta` won't satisfy the `1.4` requirement, it shouldn't be there...for now.

File: test/test.js [hunk 2]
```diff
@@ -1,51 +1,81 @@
...
+      const opts = getOps();
+      yield this.source(`${dir}/entry.js`).rollup(opts).target(tmp);
 
-test('bundle', async t => {
-  const data = ''
-  const options = {
-    file: {
-      base: 'entry.js',
-      dir: 'fixtures'
+      const res = yield this.$.read(`${tmp}/entry.js`, 'utf8');
```
Comment (frantic1048) [inline test/test.js:32]: Will this kind of interfaces be documented in new API Docs ? It could help a lot when writing tests for new plugins.

**Comment by frantic1048** [comment]
Thanks for your great work!

For beta users' accessibility, is it ok to just put this new version in master ? Then it's available via `npm i frantic1048/fly-rollup`. Or need I put it into a branch or make a tag ?

And when fly official release the 2.0, then I publish the new version of fly-rollup to npm ?

I'm asking these questions because I don't have much experience on version management. I'd appreciate for your instruction ><

**Comment by lukeed** [comment]
No problem!

Yes, everything will be documented.

And yes, you can merge it as is now and delay  `npm publish` until Fly has released 2.0 officially.

**Comment by frantic1048** [comment]
And last thing, any idea to make CI happy ? I find tests works well on local machine, but cannot pass on CI. There is a weird file not found error.
https://travis-ci.org/frantic1048/fly-rollup/jobs/176548505#L648

**Comment by lukeed** [comment]
If you rerun the rest it will work. I'm pretty sure it's because the tests are writing to, reading, and deleting the same destination file.

I was going to look into it more tomorrow, so feel free to wait if you want.

**Comment by frantic1048** [comment]
Thanks for the fixing, merged (っ´ω`c)♡

**Pull request closed without merging**